### ISSUE TITLE
[BUG] Dashboard and SU Compliance reports not working

### ISSUE THREAD

**Issue by freeli75** [SCCM-Zone/CM-SRS-Dashboards#5]
Dashboard and SU Compliance reports not working when CMDB and SSRS DB are not hosted on the same SQL server.

-Ioan

---
When I try to run the report I get this error:

An error has occurred during report processing. (rsProcessingAborted)										 Query execution failed for dataset 'ReportInfo'. (rsErrorExecutingCommand)											 For more information about this error navigate to the report server on the local server machine, or enable remote errors | An error has occurred during report processing. (rsProcessingAborted)										 Query execution failed for dataset 'ReportInfo'. (rsErrorExecutingCommand)											 For more information about this error navigate to the report server on the local server machine, or enable remote errors
-- | --
An error has occurred during report processing. (rsProcessingAborted)										 Query execution failed for dataset 'ReportInfo'. (rsErrorExecutingCommand)											 For more information about this error navigate to the report server on the local server machine, or enable remote errors

It works in our dev enviroment but not in our production, the only difference beetween them is that the reporting databas is not on the same server as the cm-database. I have run the prereq sql-script.

I get this error for all the reports exept the AL Alerts and SU SUP Sync Status reports.

**Comment by Ioan-Popovici** [comment]
Hi @freeli75 thanks for testing a beta version,

Yeah, the different reporting server DB location might be an issue.

We need to check the logs to make sure.

# Debug Steps

* Start PowerShell as admin
* Run `Get-Service SQLServerReportingServices | Stop-Service -Confirm` (Stop the reporting service)
* Go to the `<ReportingServerInstallPath>\Microsoft SQL Server Reporting Services\SSRS\LogFiles` and move all the files to a backup folder.
* Run `Get-Service SQLServerReportingServices | Start-Service` (Start the reporting service)
* Run one of the reports that fail
* Copy the `ReportingServicesService_<DateAndSequenceNumber>.log` to another location and sanitize it
* Post the log here or send it to hello@sccm-zone.com so I can review it

Thanks,
Ioan

**Comment by freeli75** [comment]
Here is the logfile.

/Fredrik
[ReportServerService__01_21_2020_16_27_48.log](https://github.com/SCCM-Zone/CM-SRS-Dashboards/files/4091827/ReportServerService__01_21_2020_16_27_48.log)

**Comment by Ioan-Popovici** [comment]
It seems it's not finding the  ReportServer.dbo.Catalog:

`...Query execution failed for dataset 'ReportInfo'. ---> System.Data.SqlClient.SqlException: Invalid object name 'ReportServer.dbo.Catalog'.`

I'm using this query to get the report path:

```sql
SELECT DISTINCT 
	Description
	, Path = REPLACE(Path, Name, '')
FROM ReportServer.dbo.Catalog
WHERE Name = @ReportName
```

The solution would be to specify the path manually or specify the report server path.
Let me sleep on this and I'll give you a fix tomorrow.

**Comment by Ioan-Popovici** [comment]
Hi @freeli75,

I thought about it and the best option would be to select a different `Data Source` for the
`ReportInfo` dataset. You will need to specify the report server connection string in that dataset.

I could make that a static value but then everyone not having your config will need to modify it.
When I finish the dashboard installer I'll modify that value to static, and you'll be able to specify the path at the installation step.

Let me know how that goes,
Ioan

**Comment by Ioan-Popovici** [comment]
Fixed in v2.0.0
